tests/test_configOverrides.py

```diff
@@ -0,0 +1,290 @@
+
+        # supported string conversions
+        self.checkSingleFieldOverride(field, "True", True)
+        self.checkSingleFieldOverride(field, "False", False)
+
+        # parseable but invalid input
+        with self.assertRaises(pexConfig.FieldValidationError):
+            self.checkSingleFieldOverride(field, 1)
+        with self.assertRaises(pexConfig.FieldValidationError):
+            self.checkSingleFieldOverride(field, [])
+        with self.assertRaises(pexConfig.FieldValidationError):
+            self.checkSingleFieldOverride(field, "1")
+
+        # non-parseable input
+        with self.assertRaises(RuntimeError):
+            self.checkSingleFieldOverride(field, "value")
+
+    def testSimpleValueInt(self):
+        """Test for applying value override to a int field
+        """
+        field = "fInt"
+
+        # values of supported type
+        self.checkSingleFieldOverride(field, 0, 0)
+        self.checkSingleFieldOverride(field, 100, 100)
+
+        # supported string conversions
+        self.checkSingleFieldOverride(field, "0", 0)
+        self.checkSingleFieldOverride(field, "100", 100)
+        self.checkSingleFieldOverride(field, "-100", -100)
+        self.checkSingleFieldOverride(field, "0x100", 0x100)
+
+        # parseable but invalid input
+        with self.assertRaises(pexConfig.FieldValidationError):
+            self.checkSingleFieldOverride(field, 1.0)
+        with self.assertRaises(pexConfig.FieldValidationError):
+            self.checkSingleFieldOverride(field, "1.0")
+        with self.assertRaises(pexConfig.FieldValidationError):
+            self.checkSingleFieldOverride(field, "[]")
+
+        # non-parseable input
+        with self.assertRaises(RuntimeError):
+            self.checkSingleFieldOverride(field, "value")
+
+    def testSimpleValueFloat(self):
+        """Test for applying value override to a float field
+        """
+        field = "fFloat"
+
+        # values of supported type
+        self.checkSingleFieldOverride(field, 0., 0.)
+        self.checkSingleFieldOverride(field, 100., 100.)
+
+        # supported string conversions
+        self.checkSingleFieldOverride(field, "0.", 0.)
+        self.checkSingleFieldOverride(field, "100.0", 100.)
+        self.checkSingleFieldOverride(field, "-1.2e10", -1.2e10)
+
+        # parseable but invalid input
+        with self.assertRaises(pexConfig.FieldValidationError):
+            self.checkSingleFieldOverride(field, [])
+        with self.assertRaises(pexConfig.FieldValidationError):
+            self.checkSingleFieldOverride(field, "(1, 1)")
+
+        # non-parseable input
+        with self.assertRaises(RuntimeError):
+            self.checkSingleFieldOverride(field, "value")
+
+    def testListValueStr(self):
+        """Test for applying value override to a list field
+        """
+        field = "fListStr"
+
+        # values of supported type
+        self.checkSingleFieldOverride(field, ["a", "b"], ["a", "b"])
+
```

Comment:
Does this implicitly support tuple if it was supplied? otherwise that might be a good test for parseable but invalid input